src/hotspot/share/runtime/stubCodeGenerator.cpp

```diff
@@ -91,11 +94,60 @@ StubCodeGenerator::~StubCodeGenerator() {
 #endif
 }
 
+void StubCodeGenerator::setup_code_desc(const char* name, address start, address end, bool loaded_from_cache) {
+  StubCodeDesc* cdesc = new StubCodeDesc("StubRoutines", name, start, end);
+  cdesc->set_disp(uint(start - _masm->code_section()->outer()->insts_begin()));
+  if (loaded_from_cache) {
+    cdesc->set_loaded_from_cache();
+  }
+  print_stub_code_desc(cdesc);
+  // copied from ~StubCodeMark()
+  Forte::register_stub(cdesc->name(), cdesc->begin(), cdesc->end());
+  if (JvmtiExport::should_post_dynamic_code_generated()) {
+    JvmtiExport::post_dynamic_code_generated(cdesc->name(), cdesc->begin(), cdesc->end());
+  }
+}
+
+// Helper used to restore ranges and handler addresses restored from
+// AOT cache. Expects entries to contain 3 * count addresses beginning
+// at offset begin which identify start of range, end of range and
+// address of handler pc. start and end of range may not be null.
+// handler pc may be null in which case it defaults to the
+// default_handler.
+
+void StubCodeGenerator::register_unsafe_access_handlers(GrowableArray<address> &entries, int begin, int count) {
```

Comment:
Is there a case or is it possible that `begin` is non -zero? I think it is same to assume we need to register all the values in `entries` in the `UnsafeMemoryAccess::_table`, right?